test/facets/Token/behaviors/bridgeToken.t.sol

```diff
@@ -0,0 +1,198 @@
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity 0.8.19;
+
+import { TokenBridge } from "../TokenBridge.t.sol";
+import { ArbForkTest } from "../../../ArbForkTest.t.sol";
+import { ITokenBridge } from "../../../../contracts/facets/Token/ITokenBridge.sol";
+
+/// @title TestBridgeToken
+/// @notice This contract tests the functionalities of the bridgeToken function
+/// @dev It inherits from the TokenWithBridge contract and the ArbForkTest contract
+/// @dev For chaging fork it is sufficient to twick the ForkTest inheritance
+contract TestBridgeToken is ArbForkTest, TokenBridge {
+    error TokenBridge__UnsupportedChain();
+    error TokenBridge__NoZeroAmount();
+    error TokenBridge__InsufficientBalance();
+    error TokenBridge__NotEnoughGas();
+
+    /// @notice Event emitted by Axelar Gateway when a contract call is made properly
+    event ContractCall(
+        address indexed sender,
+        string destinationChain,
+        string destinationContractAddress,
+        bytes32 indexed payloadHash,
+        bytes payload
+    );
+
+    event TokenBridgeInitialised(
+        string indexed destinationChain,
+        string indexed destinationAddress,
+        uint256 indexed amount
+    );
+
+    /// @dev AxelarGatway selector for approving contract calls in events
+    bytes32 internal constant SELECTOR_APPROVE_CONTRACT_CALL =
+        keccak256("approveContractCall");
+
+    /// @dev an example of a supported chain
+    string public supportedChain = "ethereum";
+    /// @dev an example of a destination address, it is a random address
+    string public destinationAddress =
+        "0x6513Aedb4D1593BA12e50644401D976aebDc90d8";
+    address public ALICE = makeAddr("Alice");
+    address public OWNER = makeAddr("Owner");
+    uint256 public ALICE_BALANCE = 100 ether;
+    /// @dev it Alice balance after claiming
+    uint256 public actualAliceBalance;
+    /// @dev address of the token proxy contract
+    address public tokenAddress;
+
+    function setUp() public virtual override {
+        vm.startPrank(OWNER);
+        super.setUp();
+        initTokenBridge(ARBITRUM_AXELAR_GATEWAY, ARBITRUM_AXELAR_GAS_SERVICE);
+        tokenAddress = address(token);
+        token.addMintingContract(address(token));
+        vm.stopPrank();
+
+        vm.startPrank(MINTER);
+        token.mint(ALICE, ALICE_BALANCE);
+        vm.stopPrank();
+
+        vm.startPrank(ALICE);
+        token.claim();
+        actualAliceBalance = token.balanceOf(ALICE);
+
+        vm.deal(ALICE, 100 ether);
+    }
+
+    /// @notice This function is used to test the onlySupportedChains modifier
+    /// @dev It tests if bridging on an unsupported chain should not be supported
+    function test_bridgingOnUnsupportedChainShouldNotBeSupported() public {
```

Comment:
Follow this pattern for all tests:

`test_functionName_whatIsBeingTested_conditionIfApplicable`.

Like this:

`test_bridgeToken_revertsWhen_bridgingToUnsupportedChain`

Reply:
fixed here https://github.com/giovannifranchi/insrt-v2-contracts/commit/63acdaccf6b616e97578b7d62e1f167b6ac8d531